topology/mask.h

```diff
@@ -312,6 +312,70 @@ class BallMask : public Mask< D >
 };
 
 /**
+ * Mask defining a elliptical region. Only 2D is implemented
+ */
+template < int D >
+class EllipseMask : public Mask< D >
+{
+public:
+  /**
+   * @param center Center of ellipse
+   * @param x_side Length between center and x vertex
+   * @param y_side Length between center and y vertex
+   */
+  EllipseMask( Position< D > center, double x_side, double y_side )
```

Comment:
@heplesser Thanks for your comments! We called the parameter names x_side and y_side because when we check if a point is inside the ellipse (in `EllipseMask< D >::inside`) we use the ellipse equation (x-x_center)^2 / x_side^2 + (y-y_center)^2 / y_side^2 <= 1, and it is important that we use the x-axis radius when dividing with x and same with y. I am unsure how this should be done with major/minor axis, because we then no longer know if the major axis comes from the x-axis or the y-axis. Do you have any suggestions as to how I could deal with this?